cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -1030,6 +1031,32 @@ class AsofJoinNode : public ExecNode {
     return match.indices()[0];
   }
 
+  static Result<std::vector<col_index_t>> GetIndicesOfOnKey(
+      const std::vector<std::shared_ptr<Schema>>& input_schema, const FieldRef& on_key) {
+    size_t n_input = input_schema.size();
+    std::vector<col_index_t> indices_of_on_key(n_input);
+    for (size_t i = 0; i < n_input; ++i) {
+      ARROW_ASSIGN_OR_RAISE(indices_of_on_key[i],
+                            FindColIndex(*input_schema[i], on_key, "on"));
+    }
+    return indices_of_on_key;
+  }
+
+  static Result<std::vector<std::vector<col_index_t>>> GetIndicesOfByKey(
```

Comment:
Same comment as `GetIndicesOfByKey`

Reply:
Sorry, which comment? You're referring to `GetIndicesOfByKey` but it's here.